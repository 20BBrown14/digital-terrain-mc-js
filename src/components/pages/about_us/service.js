import axios from 'axios';
import ServiceValidationError from '../../../modules/ServiceValidationError';

const ABOUT_US_PATH = '/aboutUsInformation';
const VETERANS_PATH = '/veteransInformation';

const retrieveAboutUsInformation = (successCallback, failureCallback) => {
  axios.get(ABOUT_US_PATH)
    .then((response) => {
      const responseJson = response.data;
      if (!responseJson) {
        throw new ServiceValidationError('Missing response data');
      }

      successCallback(responseJson);
    })
    .catch(failureCallback);
};

const retrieveVeteransInformation = (successCallback, failureCallback) => {
  axios.get(VETERANS_PATH)
    .then((response) => {
      const responseJson = response.data;
      if (!responseJson) {
        throw new ServiceValidationError('Missing response data');
      }

      successCallback(responseJson);
    })
    .catch(failureCallback);
};

export default { retrieveAboutUsInformation, retrieveVeteransInformation };
