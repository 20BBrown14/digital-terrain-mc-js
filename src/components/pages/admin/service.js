import axios from 'axios';
import DataSaveError from '../../../modules/DataSaveError';
import UnexpectedDataError from '../../../modules/UnexpectedDataError';

const SAVE_JSON_PATH = '/save';
const LOAD_APPS_PATH = '/loadApps';
const UPDATE_APP_STATUS_PATH = '/updateappstatus';
const DELETE_APP_PATH = '/deleteapp';

export const saveJSONInformationService = (successCallback, failureCallback, JSONTypeToSave, JSONToSave) => {
  axios.post(
    `${SAVE_JSON_PATH}?JSONTypeToSave=${JSONTypeToSave}`,
    {
      JSON: JSONToSave,
    },
    {
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    },
  )
    .then((response) => {
      const responseJson = response.data;
      if (response.status > 499) {
        throw new DataSaveError('Data failed to save. Please try again later');
      } else if (response.status > 399 && response.status <= 499) {
        throw new UnexpectedDataError('Data or request is illformed. Please try again.');
      } else if (response.status > 299) {
        throw new Error('Unexpected Error while attempting to save data. Please try again later.');
      }

      successCallback(responseJson);
    })
    .catch(failureCallback);
};

export const loadAppsService = (successCallback, failureCallback, applicationFilter) => {
  axios.get(
    `${LOAD_APPS_PATH}?applicationFilter=${applicationFilter}`,
  )
    .then((response) => {
      const responseData = response.data;
      if (responseData) {
        successCallback(responseData);
      }
    })
    .catch(failureCallback);
};

export const updateAppStatusService = (successCallback, failureCallback, appID, newStatus) => {
  axios.post(
    UPDATE_APP_STATUS_PATH,
    {
      appID,
      newStatus,
    },
    {
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    },
  )
    .then(successCallback)
    .catch(failureCallback);
};

export const deleteAppService = (successCallback, failureCallback, appID) => {
  axios.post(
    DELETE_APP_PATH,
    {
      appID,
    },
    {
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    },
  )
    .then(successCallback)
    .catch(failureCallback);
};
