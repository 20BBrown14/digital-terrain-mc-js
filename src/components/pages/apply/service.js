import axios from 'axios';

const APPLICATION_PATH = '/applicationsubmit';

const submitApplicationService = (successCallback, failureCallback, applicationInformation) => {
  axios.post(
    APPLICATION_PATH,
    {
      applicationInformation,
    },
    {
      headers: {
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
    },
  )
    .then(successCallback)
    .catch(failureCallback);
};

export default submitApplicationService;
