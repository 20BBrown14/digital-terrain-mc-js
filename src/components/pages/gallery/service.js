import axios from 'axios';
import ServiceValidationError from '../../../modules/ServiceValidationError';

const RETRIEVE_IMAGES_PATH = '/galleryImages';

const retrieveGalleryImages = (successCallback, failureCallback, isFeatured = false) => {
  axios.get(`${RETRIEVE_IMAGES_PATH}?isFeatured=${isFeatured}`)
    .then((response) => {
      const responseJson = response.data;
      if (!responseJson) {
        throw new ServiceValidationError('Missing response data');
      }

      successCallback(responseJson);
    })
    .catch(failureCallback);
};

export default retrieveGalleryImages;
