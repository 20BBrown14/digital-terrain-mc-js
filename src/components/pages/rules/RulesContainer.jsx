import React from 'react';
import RulesView from './RulesView';

/**
 * Rules page container.
 */
const RulesContainer = () => (
  <RulesView />
);

export default RulesContainer;
