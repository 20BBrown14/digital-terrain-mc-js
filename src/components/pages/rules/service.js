import axios from 'axios';
import ServiceValidationError from '../../../modules/ServiceValidationError';

const RULES_PATH = '/serverRules';

const retrieveRulesInformation = (successCallback, failureCallback) => {
  axios.get(RULES_PATH)
    .then((response) => {
      const responseJson = response.data;
      if (!responseJson) {
        throw new ServiceValidationError('Missing response data');
      }

      successCallback(responseJson);
    })
    .catch(failureCallback);
};

export default retrieveRulesInformation;
