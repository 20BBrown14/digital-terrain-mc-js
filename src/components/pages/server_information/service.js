import axios from 'axios';
import ServiceValidationError from '../../../modules/ServiceValidationError';

const SERVER_INFORMATION_PATH = '/serverInformation';

const retrieveServerInformation = (successCallback, failureCallback) => {
  axios.get(SERVER_INFORMATION_PATH)
    .then((response) => {
      const responseJson = response.data;
      if (!responseJson) {
        throw new ServiceValidationError('Missing response data');
      }

      successCallback(responseJson);
    })
    .catch(failureCallback);
};

export default retrieveServerInformation;
