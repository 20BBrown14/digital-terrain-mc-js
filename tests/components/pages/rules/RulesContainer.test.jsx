import React from 'react';
import RulesContainer from '../../../../src/components/pages/rules/RulesContainer';
import service from '../../../../src/components/pages/rules/service';

jest.mock('../../../../src/components/pages/rules/service');

describe('RulesContainer', () => {
  afterEach(() => {
    service.mockClear();
  });

  it('renders a default view', () => {
    const container = shallow(<RulesContainer />);
    container.instance().state.isLoading = false;
    container.update();
    expect(container).toMatchSnapshot();
  });

  it('has a RulesView', () => {
    const container = shallow(<RulesContainer />);
    expect(container.find('RulesView')).toHaveLength(1);
  });

  describe('componentDidMount()', () => {
    it('calls service', () => {
      const container = shallow(<RulesContainer />).instance();
      service.mockClear();
      container.componentDidMount();
      expect(service).toHaveBeenCalledTimes(1);
    });

    describe('successful service', () => {
      it('sets state as expected', () => {
        const container = shallow(<RulesContainer />).instance();
        service.mockImplementation((success) => {
          success({ rules: [] });
        });
        container.state.isLoading = true;
        container.componentDidMount();

        expect(container.state.rulesInformation).toEqual({ rules: [] });
        expect(container.state.isLoading).toEqual(false);
      });
    });

    describe('failed service', () => {
      it('sets state as expected', () => {
        const container = shallow(<RulesContainer />).instance();
        service.mockImplementation((success, failed) => {
          failed();
        });
        container.state.isLoading = true;
        container.componentDidMount();

        expect(container.state.hasServiceFailure).toEqual(true);
        expect(container.state.isLoading).toEqual(false);
      });
    });
  });
});
