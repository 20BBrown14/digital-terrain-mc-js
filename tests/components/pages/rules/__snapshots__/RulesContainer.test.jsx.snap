// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RulesContainer renders a default view 1`] = `
<div
  class="rules-page-content"
>
  <h1
    class="ant-typography rules-page-title"
  >
    Digital Terrain Rules and Guidelines
  </h1>
  <div
    class="ant-collapse ant-collapse-icon-position-left"
  >
    <div
      class="ant-collapse-item ant-collapse-item-active"
    >
      <div
        aria-expanded="true"
        class="ant-collapse-header"
        role="button"
        tabindex="0"
      >
        <span
          aria-label="right"
          class="anticon anticon-right ant-collapse-arrow"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="right"
            fill="currentColor"
            focusable="false"
            height="1em"
            style="-ms-transform:rotate(90deg);transform:rotate(90deg)"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
            />
          </svg>
        </span>
        <h1>
          General Guidelines
        </h1>
      </div>
      <div
        class="ant-collapse-content ant-collapse-content-active"
      >
        <div
          class="ant-collapse-content-box"
        >
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              1. Be considerate of everybody on the server by being conscious of how you interact with others.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              2. AFKing is allowed for automated farms. Automated character movement or entity detection is disallowed.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              3. You may be kicked while AFK if tps takes a dip.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              4. Do not build within ~200 blocks of another player's claimed without permission.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              5. Don't build in spawn chunks without permission.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              6. No use of mods that give you an unfair advantage (x-ray, vanilla x-ray bugs).
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              7. The Discord server should be used when you're wanting to claim a limited structure (witch hut, fortress, bastion, etc).
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              8. You should not alter community builds or spaces without permission.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              9. Do not spam chat in-game or Discord. This includes using all caps for more than a few messages.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              10. Do not grief or steal as you will be found out and able to be fixed in 10 seconds.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              11. PVP is allowed but don't be aggressive or abusive
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              12. Your Discord nickname should match your in-game name exactly.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              13. Do not advertise other Minecraft servers in-game or in Discord.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              14. Being in the Discord is a requirement for being whitelisted. If you leave you will automatically be un-whitelisted.
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="ant-collapse-item ant-collapse-item-active"
    >
      <div
        aria-expanded="true"
        class="ant-collapse-header"
        role="button"
        tabindex="0"
      >
        <span
          aria-label="right"
          class="anticon anticon-right ant-collapse-arrow"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="right"
            fill="currentColor"
            focusable="false"
            height="1em"
            style="-ms-transform:rotate(90deg);transform:rotate(90deg)"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
            />
          </svg>
        </span>
        <h1>
          Redstone Guidelines
        </h1>
      </div>
      <div
        class="ant-collapse-content ant-collapse-content-active"
      >
        <div
          class="ant-collapse-content-box"
        >
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              1. No lag machines.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              2. Cover hoppers with composters.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              3. Light redstone wires.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              4. All machines need an off switch.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              5. All machines need overflow protection that turns off production of entities or shoots excess into lava.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              6. TNT duping is allowed with permission.
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="ant-collapse-item ant-collapse-item-active"
    >
      <div
        aria-expanded="true"
        class="ant-collapse-header"
        role="button"
        tabindex="0"
      >
        <span
          aria-label="right"
          class="anticon anticon-right ant-collapse-arrow"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="right"
            fill="currentColor"
            focusable="false"
            height="1em"
            style="-ms-transform:rotate(90deg);transform:rotate(90deg)"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
            />
          </svg>
        </span>
        <h1>
          Shop Guidelines
        </h1>
      </div>
      <div
        class="ant-collapse-content ant-collapse-content-active"
      >
        <div
          class="ant-collapse-content-box"
        >
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              1. The shopping district is only for shops.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              2. Spawn proof your shops and community areas. /ll can help you.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              3. Abandoned shops will be dismantled and all materials and stock donated to the community.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              4. Claiming shopping district space costs is 1 diamond block for 100 blocks of x/z space rounded up.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              5. Prosperity Plaza Mall space is free to claim if you don't want a standalone shop. See mall specific guidelines.
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="ant-collapse-item ant-collapse-item-active"
    >
      <div
        aria-expanded="true"
        class="ant-collapse-header"
        role="button"
        tabindex="0"
      >
        <span
          aria-label="right"
          class="anticon anticon-right ant-collapse-arrow"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="right"
            fill="currentColor"
            focusable="false"
            height="1em"
            style="-ms-transform:rotate(90deg);transform:rotate(90deg)"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
            />
          </svg>
        </span>
        <h1>
          Prosperity Plaza Mall Guidelines
        </h1>
      </div>
      <div
        class="ant-collapse-content ant-collapse-content-active"
      >
        <div
          class="ant-collapse-content-box"
        >
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              1. space is free to claim for anybody. Claim a space by renaming the paper in the directory with the space you want with your shop name.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              2. Completely spawn proof your claimed space
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              3. You can customize two blocks in on the side walls, ceiling, and floor. You are allowed to customize the backwall as you please as many blocks as you please.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              4. You are not allowed to customize the mall facing wall EXCEPT to add/adjust a door or window.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              5. Any default blocks removed must be returned to mall office.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              6. Abandoned shops will be removed and space back up for claim
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="ant-collapse-item ant-collapse-item-active"
    >
      <div
        aria-expanded="true"
        class="ant-collapse-header"
        role="button"
        tabindex="0"
      >
        <span
          aria-label="right"
          class="anticon anticon-right ant-collapse-arrow"
          role="img"
        >
          <svg
            aria-hidden="true"
            class=""
            data-icon="right"
            fill="currentColor"
            focusable="false"
            height="1em"
            style="-ms-transform:rotate(90deg);transform:rotate(90deg)"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
            />
          </svg>
        </span>
        <h1>
          Community Farm Guidelines
        </h1>
      </div>
      <div
        class="ant-collapse-content ant-collapse-content-active"
      >
        <div
          class="ant-collapse-content-box"
        >
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              1. Community farm locations can be found on the Server Information tab.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              2. Community farms have a strict 'no extended afk' policy. If you are afk another player may come and use the farm.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              3. Only take resources that you farm. Don't just always take any backstock.
            </span>
          </p>
          <p>
            <span
              class="ant-typography rules-page-text"
            >
              4. You are allowed to sell any materials you get from a community farm.
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
